lib/base/json.cpp

```diff
@@ -450,14 +450,18 @@ template<bool prettyPrint>
 inline
 String JsonEncoder<prettyPrint>::GetResult()
 {
-	return String(m_Result.begin(), m_Result.end());
+	std::string result;
+
+	std::swap(result, m_Result);
+
+	return std::move(result);
```

Comment:
What's the reason for this combination of swap and move?

Reply:
> What's the reason for this combination of swap and move?

I wanted to leave the instance in a defined state while extracting the result, hence I exchanged the latter with an empty string. But as JsonEncoder is json.cpp-only anyway, that shouldn't be necessary...